test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+
+//! @brief Standard history data management object.
+//! @tparam T: Data type of the history data
+template <typename T>
+class HistoryData
+{
+    std::map<int, std::vector<T>> mHisData;
+
+public:
+    //! @brief Returns the history data for a specific integration point
+    //! @param cellNum: Number of the cell containing the integration point
+    //! @param ipNum: Number of the integration point
+    //! @return History data of the integration point
+    const T& GetIPHistoryData(int cellNum, int ipNum) const
+    {
+        auto ipVecIt = mHisData.find(cellNum);
+        if (ipVecIt == mHisData.end())
+            throw Exception(__PRETTY_FUNCTION__, "Cell data not found! - Did you initialize the history data?");
+        return (*ipVecIt).second[ipNum];
+    }
+
+    //! @brief Returns the history data for a specific integration point
+    //! @param cellNum: Number of the cell containing the integration point
+    //! @param ipNum: Number of the integration point
+    //! @return History data of the integration point
+    T& GetIPHistoryData(int cellNum, int ipNum)
+    {
+        auto ipVecIt = mHisData.find(cellNum);
+        if (ipVecIt == mHisData.end())
+            throw Exception(__PRETTY_FUNCTION__, "Cell data not found! - Did you initialize the history data?");
+        return (*ipVecIt).second[ipNum];
+    }
+
+    //! @brief Initializes the history data
+    //! @param cells: Cellgroup that is related to the integrand
+    //! @param dof: Dof type
+    void InitializeHistoryData(const Groups::Group<CellInterface>& cells, DofType dof)
+    {
+        for (CellInterface& cell : cells)
+        {
+            cell.Apply([this](const NuTo::CellData& cellData, const NuTo::CellIpData& cellIpData) {
+                if (mHisData.find(cellData.GetCellId()) == mHisData.end())
+                    mHisData.emplace(cellData.GetCellId(), std::vector<T>(cellData.GetNumIntegrationPoints()));
+            });
+        }
+    }
+};
+
+
+// %%% Custom law with history data %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+//! @brief Structure that holds the history data of the creep law
+struct CreepHistoryData
+{
+    EngineeringStrainPDE<1> prevStrain;
+    EngineeringStressPDE<1> prevStress;
+    VectorXd gamma = Eigen::VectorXd::Zero(2);
+
+    CreepHistoryData()
+    {
+        prevStrain[0] = 0.;
+        prevStress[0] = 0.;
+    }
+};
+
+//! @brief Creep law using the exponential algorithm
+class CreepLaw : public Laws::MechanicsInterface<1>, public HistoryData<CreepHistoryData>
+{
+
+    double mE; //!< Youngs modulus
+    VectorXd mE_KC; //!< Kelvin chain stiffness
+    VectorXd mT_KC; //!< Kelvin chain retardation time
+
+
+public:
+    using typename Laws::MechanicsInterface<1>::MechanicsTangent;
+
+
+    //! @brief Ctor
+    //! @param E: Youngs modulus
+    //! @param E_KC: Kelvin chain stiffness
+    //! @param T_KC: Kelvin chain retardation time
+    CreepLaw(double E, VectorXd E_KC, VectorXd T_KC)
```

Comment:
For a first pull request an a discussion of the idea, I guess that a creep law with some time integration is not the optimal example. A 1D-plasticity model is very simple and easier to understand.

Reply:
As I already told TTitscher: for me it was more than just writing a test. It was about understanding PDE and I am much more familiar to creep than to plasticity. In the OP, I removed all the creep related stuff and only left the things that are relevant to history data. It is also my intention to move this  to the examples later. I am not as familiar with the PDE implementations as you guys. So I wasn't able to write a minimal test. But as soon as I am, I ll do that.